Monte_carlo.cpp

```diff
@@ -0,0 +1,324 @@
+            transportNeutron(bounds, tallies, first_round,
+                    mesh, old_fission_bank, new_fission_bank);
+        }
+
+        // give results
+        std::cout << "For batch " << batch << ", k = "
+            << tallies[FISSIONS].getCount()/(tallies[LEAKS].getCount() +
+                    tallies[ABSORPTIONS].getCount()) << std::endl;
+        first_round = false;
+    }
+    double mean_crow_distance = tallies[CROWS].getCount()
+        / tallies[NUM_CROWS].getCount();
+    std::cout << "Mean crow fly distance = " << mean_crow_distance;
+
+    delete old_fission_bank;
+    delete new_fission_bank;
+}
+
+/*
+ @brief     Function that generates a neutron and measures how 
+            far it travels before being absorbed.
+ @details   A neutron is created in the bounding box using
+            sample_location() for the first batch and sample_fission_site()
+            for the rest of the batches. It moves a distance determined by
+            sample_distance(). It is then either absorbed or
+            scattered as determined by sample_interaction(). When
+            it is absorbed, its distance from its starting point
+            is appended to crow_distances. If the absorption creates
+            a fission event, the number of neutrons emited is sampled.
+            The location of the fission event is added to a list of fission
+            events.
+ @param	    mat a Material object containing information about the material
+ @param     bounds a Boundaries object containing the limits
+            of the bounding box
+ @param     tallies a dictionary containing tallies of crow distances,
+            leakages, absorptions, and fissions
+ @param     fission_banks a dictionary containing the old and new fission banks
+ @param     first_round a boolean telling whenther or not this is the first
+            batch to be tested
+ @param     mesh a Mesh object containing information about the mesh
+*/
+void transportNeutron(Boundaries bounds, std::vector <Tally> &tallies,
+        bool first_round, Mesh &mesh,
+        std::vector <std::vector <double> >* old_fission_bank,
+        std::vector <std::vector <double> >* new_fission_bank) {
+    
+    // declare variables
+    const double TINY_MOVE = 1e-10;
+    double theta;
+    double phi;
+    double neutron_distance;
+    double tempd;
+    double r;
+    double bound_val;
+    double crow_distance;
+    int group;
+    int new_group;
+    int neutron_interaction;
+    std::vector <double> neutron_starting_point;
+    std::vector <double> neutron_position;
+    std::vector <double> neutron_direction;
+    std::vector <double> cell_mins;
+    std::vector <double> cell_maxes;
+    std::vector <std::vector <double> > distance_to_cell_edge;
+    std::vector <double> temp_sigma_s_group;
+    std::vector <int> cell;
+    std::vector <int> cell_lim_bound;
+    std::vector <int> box_lim_bound;
+    bool neutron_lost = false;
+    Material* cell_mat;
```

Comment:
So...this is an old school way to declare variables. It's actually enforced in older languages like FORTRAN. But in C/C++ it's actually better to define your variable around the area where you use it. It allows for easier compiler optimizations at times and it also improves readability.